packages/docusaurus-plugin-content-docs/src/plugin-content-docs.d.ts

```diff
@@ -404,6 +406,8 @@ declare module '@docusaurus/plugin-content-docs' {
     tags: Tag[];
     /** Front matter, as-is. */
     frontMatter: DocFrontMatter & {[key: string]: unknown};
+    /** Is the doc going to be excluded from production builds? */
+    isDraft: boolean;
```

Comment:
I'd prefer to keep consistent naming, e.g. if front matter is called `draft` it should still be `draft` here. Also, per my other comment, if we filter out draft docs first, we don't need to have it in metadata at all.